### PR TITLE
Feat/show disconnect

### LINKED ISSUE #339: Surface web socket connection failure to user

### PR DESCRIPTION
**Pull request by Kevingislason**
## Description

Closes #339

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [x] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [x] Updated relevant documentation in the code
- [x] Test code changes manually, and describe your procedure in a comment on this PR
- [x] Mark this PR with the correct milestone

### CODE REVIEW

File: src/contexts/substrateContext.tsx [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { ApiPromise, WsProvider } from '@polkadot/api';
 import types from '../config/types.json';
 import { useConfig } from './configContext';
 
+export const API_STATE = {
```
Comment (zhangjing28) [inline src/contexts/substrateContext.tsx:8]: Nit: if you don't care about the value of `API_STATE`, you can use 
```
export enum API_STATE = {
  CONNECT_INIT,
  READY,
  ERROR,
  DISCONNECTED
};
```
It saves you a little effort.

File: src/utils/display/getBalanceText.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { bool } from '@polkadot/types';
```
Comment (zhangjing28) [inline src/utils/display/getBalanceText.ts:1]: This file has the same codes as `getZkTransactBalanceText ` file. Is that make sense?

**Review by zhangjing28 — APPROVED** [review]

**Review by Boyuan-Deng — APPROVED** [review]
Great refactor on BridgeBalanceInput & SendBalanceInput. Good effort on integrating Typescript into Balance. Overall code quality looks pretty good and thinks the UI for handling bridge api disconnect makes sense.

File: src/components/Accounts/ZkAccountModal.tsx [hunk 3]
```diff
@@ -41,38 +79,39 @@ const PrivateTokenItem = ({ balance }) => {
...
-          </div>
+    <div className="border border-secondary rounded-lg px-6 py-6 text-secondary overflow-y-auto bg-white bg-opacity-5">
+      <div className="flex flex-col items-center justify-center gap-3">
+        <div className="text-white text-center">
+        Cannot connect to the network
         </div>
-        <div className="flex flex-col border border-secondary rounded-lg px-6 py-4 mt-3 text-secondary overflow-y-auto h-48 bg-white bg-opacity-5">
-          <PrivateTokenTableContent balances={balances} />
+        <div className="text-secondary text-xss">
+        Please check your internet connection and wait to reconnect.
```
[the remark this answers is not in the document]
Comment (Kevingislason) [reply src/components/Accounts/ZkAccountModal.tsx:90]: I've found that it will reconnect without having to refresh the page

**Pull request merged**